pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -164,9 +164,32 @@ func commonArgs(cluster string, cfg *config.Cluster) ([]string, error) {
 	if usernsRemap() {
 		args = append(args, "--userns=host")
 	}
+
+	// handle Docker on Btrfs or ZFS
+	dockerStorage, err := getDockerStorage()
```

Comment:
in the future this should be refactored to get all the docker info we want, we should avoid unnecessarily making extra info calls.